app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/FileItem.java

```diff
@@ -0,0 +1,94 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.net.Uri;
+import android.os.Parcel;
+import android.os.Parcelable;
+import androidx.annotation.IntDef;
+
+import androidx.annotation.NonNull;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.FileItem.FileStatus.*;
+
+/**
+ * Helper class, part of the local file sharing module.
+ *
+ * Defines a file-item to represent the files being transferred.
+ */
+public class FileItem implements Parcelable {
+
+  @Retention(RetentionPolicy.SOURCE)
+  @IntDef({ TO_BE_SENT, SENDING, SENT, ERROR })
+  public @interface FileStatus {
+    int TO_BE_SENT = -1;  // File yet to be sent
+    int SENDING = 0;      // Being sent
+    int SENT = +1;        // Successfully sent
+    int ERROR = +2;       // Error encountered while transferring the file
+  }
+
+  private Uri fileUri;
+  private String fileName;
+  private int fileStatus;
+
+  public FileItem(@NonNull Uri fileUri) { // For sender devices
```

Comment:
There should always be a constructor that just sets the fields and is you have multiple constructors they should call it with `this(field1,field2,field3)` I believe I left an example in a comment. It can be private if you never intend for that constructor to be invoked externally

Reply:
Thanks for the design tip
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/4b260e15ed27aa4f9329826ca90da5c4a81e51f3#diff-3e6b385066051312aca0978588d4db39R42)